codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -898,12 +966,30 @@ let rec checkBaseRegisterContainsRegisteredContract = (
 Recurses through registers and determines whether a contract has already been registered with
 the given name and address
 */
-let checkContainsRegisteredContractAddress = (self: t, ~contractName, ~contractAddress) => {
-  self.baseRegister->checkBaseRegisterContainsRegisteredContract(~contractName, ~contractAddress) ||
+let checkContainsRegisteredContractAddress = (
+  self: t,
+  ~contractName,
+  ~contractAddress,
+  ~chainId,
+) => {
+  self.baseRegister->checkBaseRegisterContainsRegisteredContract(
+    ~contractName,
+    ~contractAddress,
+    ~chainId,
+  ) ||
     self.pendingDynamicContracts->Array.some(({dynamicContracts}) =>
-      dynamicContracts->Array.some(dcr =>
-        dcr.contractAddress == contractAddress && (dcr.contractType :> string) == contractName
-      )
+      dynamicContracts->Array.some(dcr => {
+        let exists = dcr.contractAddress == contractAddress
+        if exists {
+          warnIfAttemptedAddressRegisterOnDifferentContracts(
+            ~contractAddress,
+            ~contractName,
+            ~existingContractName=(dcr.contractType :> string),
+            ~chainId,
+          )
+        }
```

Comment:
Will it ever reach this code with an exception thrown from `addAddress`?

Reply:
Yes it can because the error thrown will only occur for an individual contractAddressMapping when the address actually gets added or if two mappings are combined etc.

There is no global contractAddressMapping.

What was happening before is `checkContractIsRegistered` would return false in the case that the address is exists but the contractName does not match. Now it returns true so it stops it from overwriting but there is no feedback to the user that this is happening and no way for us to trace it without a log. We don't need the log but it can be helpful for debugging.